### PR TITLE
[IMPROVE][Performance] Add new database indexes to improve data query performance

### PR DESCRIPTION
**Pull request by rodrigok**
## Proposed changes

Add some missing indexes on:
* Livechat Visitors
* Messages
* Rooms
* Subscriptions
* Sessions
* _Trash

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/models/server/models/_BaseDb.js [hunk 1]
```diff
@@ -10,11 +10,13 @@ const baseName = 'rocketchat_';
...
 const trash = new Mongo.Collection(`${ baseName }_trash`);
 try {
-	trash._ensureIndex({ collection: 1 });
+	trash._ensureIndex({ __collection__: 1 });
 	trash._ensureIndex(
 		{ _deletedAt: 1 },
 		{ expireAfterSeconds: 60 * 60 * 24 * 30 },
 	);
+
+	trash._ensureIndex({ __collection__: 1, _deletedAt: 1, rid: 1 });
```
Comment (sampaiodiego) [inline app/models/server/models/_BaseDb.js:19]: I didn't find a query where `rid` is used along with `__collection__` and `_deletedAt`.. I think the best index would be without `rid`

File: app/models/server/models/Rooms.js [hunk 2]
```diff
@@ -11,12 +11,17 @@ export class Rooms extends Base {
...
+		this.tryEnsureIndex({ muted: 1 }, { sparse: true });
 		this.tryEnsureIndex({ t: 1 });
 		this.tryEnsureIndex({ 'u._id': 1 });
-		this.tryEnsureIndex({ 'tokenpass.tokens.token': 1 });
 		this.tryEnsureIndex({ ts: 1 });
+		// Tokenpass
+		this.tryEnsureIndex({ 'tokenpass.tokens.token': 1 }, { sparse: true });
+		this.tryEnsureIndex({ tokenpass: 1 }, { sparse: true });
+		// livechat
+		this.tryEnsureIndex({ 'v.token': 1 }, { sparse: true });
```
Comment (renatobecker) [inline app/models/server/models/Rooms.js:24]: We have been adding livechat-related indexes to the `LivechatRoom` model, I think it's the case for the `v.token` field.

File: app/models/server/models/_BaseDb.js (hunk 1 above)
Comment (rodrigok) [reply app/models/server/models/_BaseDb.js:19]: It's used [here](https://github.com/RocketChat/Rocket.Chat/blob/b7419757d65097742951a6ddaf37b272259c56da/server/publications/room/emitter.js#L9) calling the `trashFind` which receives a query, we found this trough database profiling

Comment (sampaiodiego) [reply app/models/server/models/_BaseDb.js:19]: oh.. ok, that's hard to track 😞 

but I've [found recently](https://docs.mongodb.com/manual/core/index-compound/#compound-index-prefix) that field order is important to compound indexes.. so in this case, I think the proper order would be:
```suggestion
	trash._ensureIndex({ rid: 1, __collection__: 1, _deletedAt: 1 });
```

so even this index will be used if do not include `_deletedAt_` on your query.

Comment (rodrigok) [reply app/models/server/models/_BaseDb.js:19]: Makes sense

**Pull request merged**